Automattic/jetpack
# Fix/waf generate rules files while api call fails

This ensures that the `rules.php` is still generated if the API request fails. It is only generated if the file does not yet exist, in order to not destroy an already good file and remove all its rules when there is no need to do so.

#### Changes proposed in this Pull Request:

* If the API request fails and the `rules.php` does not yet exist, it gets generated to include the IP allow/block lists.
* If the API request fails and the `rules.php` does already exist, it is left alone.

#### Jetpack product discussion



1201069996155224-as-1202169135150822

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






Currently the API request fails always, so this is easy to test
* Check out this branch
* If you have sandboxed your API before, make sure to undo that so that the request does indeed fail
* Ensure that you have an existing rules.php (i.e. take the one from before checking out this branch, as it gets deleted in here)
* Make a change to your existing rules.php that you will recognize (i.e. add a comment)
* Change settings in the UI and save -> error still occurs and the rules.php file should be left alone. Also, the IP config changes should still land in `allow-ip.php` or `block-ip.php` respectively.
* Remove the `rules.php`
* Change settings in the UI and save -> error still occurs but the rules.php should be regenerated (despite with no rules) and include the block/allow lists. Changes to the IP lists should still land in their respective files.
